packages/smooth_app/lib/pages/folksonomy/folksonomy_page.dart

```diff
@@ -0,0 +1,187 @@
+                      labelText: appLocalizations.tag_key,
+                      errorText: isValidKey
+                          ? null
+                          : appLocalizations.invalid_key_format,
+                    ),
+                    enabled: oldKey == null,
+                  ),
+                  TextField(
+                    controller: valueController,
+                    decoration:
+                        InputDecoration(labelText: appLocalizations.tag_value),
+                  ),
+                ],
+              ),
+              negativeAction: SmoothActionButton(
+                onPressed: () {
+                  Navigator.of(context).pop();
+                },
+                text: appLocalizations.cancel,
+              ),
+              positiveAction: SmoothActionButton(
+                onPressed: isValidKey
+                    ? () async {
+                        if (oldKey == null) {
+                          await provider.addTag(
+                              keyController.text, valueController.text);
+                        } else {
+                          await provider.editTag(oldKey, valueController.text);
+                        }
+                        if (context.mounted) {
+                          Navigator.of(context).pop();
+                        }
+                      }
+                    : null,
+                text: appLocalizations.save,
+              ),
+            );
+          },
+        );
+      },
+    );
+  }
+
+  void _showBottomSheet(BuildContext context, AppLocalizations appLocalizations,
+      FolksonomyProvider provider,
+      {required String key, required String value, String? comment}) {
+    showSmoothModalSheet(
+      context: context,
+      builder: (BuildContext context) {
+        return Wrap(
+          children: <Widget>[
+            ListTile(
+              leading: const Icon(Icons.edit),
+              title: Text(appLocalizations.edit_tag),
+              onTap: () {
+                Navigator.pop(context);
+                _showAddEditDialog(context, appLocalizations, provider,
+                    oldKey: key, oldValue: value);
+              },
+            ),
+            ListTile(
+              leading: const Icon(Icons.delete),
+              title: Text(appLocalizations.remove_tag),
+              onTap: () async {
+                Navigator.pop(context);
+                await provider.deleteTag(key);
+              },
+            ),
+          ],
+        );
+      },
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context);
+    final FolksonomyProvider provider = context.watch<FolksonomyProvider>();
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(appLocalizations.product_tags_title),
+      ),
+      body: provider.isLoading
+          ? const Center(child: CircularProgressIndicator.adaptive())
+          : ListView(
```

Comment:
Please avoid as possible `ListView` and use instead `ListView.builder`.
You will gain a lot of performance.